RHDEVDOCS 5729 overriding a task in remote pipeline (#68447)
openshift/openshift-docs: modules/op-using-remote-pipeline-annotations-with-pipelines-as-code.adoc

You can reference only one pipeline definition using the annotation.
====

[id="overriding-tasks-in-remote-pipeline_{context}"]
== Overriding a task in a remote pipeline

By default, if you use a remote pipeline annotation in a pipeline run, {pac} uses all the tasks that are a part of the remote pipeline.

> To override a task, you can add an annotation for a task with the same name to the remote pipeline annotation.

Reviewer: nit: task with the same name as in remote pipeline to the pipelinerun annotation

> To override a task, you can add an annotation for a task with the same name as a task in the pipeline that the remote pipeline annotation references.

Reviewer 2: This line is not clear to me, but I am not familiar with pipelines. Maybe something like:
```suggestion
You can override a task in a remote pipeline by adding a task that has the same name as an annotation in the pipeline run.
```

For example, you might use the following pipeline run definition:

.Example pipeline run definition referencing a remote pipeline and overriding a task
[source,yaml]
----

> kind: PipelineRun

Reviewer 3: Should this have an apiVersion here?

metadata:
  annotations:
    pipelinesascode.tekton.dev/pipeline: "https://git.provider/raw/pipeline.yaml"
    pipelinesascode.tekton.dev/task: "./my-git-clone-task.yaml"
----

> For this example, assume the remote task found at \https://git.provider/raw/pipeline.yaml includes a task named `git-clone` and the task that the `my-git-clone-task.yaml` file defines is also named `git-clone`.

Reviewer 3: the URL here should probably be in backticks
